sklearn/tests/test_base.py

```diff
@@ -359,3 +361,41 @@ def test_pickle_version_warning():
     # check that no warning is raised for external estimators
     TreeNoVersion.__module__ = "notsklearn"
     assert_no_warnings(pickle.loads, tree_pickle_noversion)
+
+
+def test_freeze():
+    X, y = datasets.load_iris(return_X_y=True)
+
+    for copy in [False, True]:
+        est = SelectKBest(k=1).fit(X, y)
+
+        frozen_est = freeze(est, copy=copy)
+        if copy:
+            assert_false(est is frozen_est)
+        else:
+            assert_true(est is frozen_est)
+        assert_array_equal(est.scores_, frozen_est.scores_)
+        assert_true(isinstance(frozen_est, SelectKBest))
+
+        dumped = pickle.dumps(frozen_est)
+        frozen_est2 = pickle.loads(dumped)
+        assert_false(frozen_est is frozen_est2)
+        assert_array_equal(est.scores_, frozen_est2.scores_)
+
+        # scores should be unaffected by new fit
+        assert_true(frozen_est2.fit() is frozen_est2)
```

Comment:
This is always true, right? Well, I guess you're testing that fit can be called without arguments?